src/main/scala/cromwell/engine/db/slick/DataAccessController.scala

```diff
@@ -0,0 +1,310 @@
+
+import java.util.{UUID, Date}
+
+import cromwell.binding.Call
+import cromwell.binding.types.WdlType
+import cromwell.binding.values.{WdlPrimitive, WdlValue}
+import cromwell.engine.db._
+import cromwell.engine.store.ExecutionStore.ExecutionStatus
+import cromwell.engine.store.SymbolStore.SymbolStoreKey
+import cromwell.engine.store.SymbolStoreEntry
+import cromwell.engine.{WorkflowId, WorkflowState, WorkflowSubmitted}
+import cromwell.util.DatabaseConfig
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ExecutionContext, Await, Future}
+import scala.language.implicitConversions
+
+// TODO: Need lots of refactoring including: better futures handling, compiled queries, utility functions, etc.
+object DataAccessController extends DataAccess {
+  val dataAccess = new DataAccessComponent(DatabaseConfig.slickDriver)
+
+  import dataAccess.driver.api._
+
+  def database: Database = Database.forConfig("", DatabaseConfig.databaseConfig)
+
+  // TODO: Pass in the execution context instead of using the global!!
+  private implicit val executionContext = ExecutionContext.global
+
+  private implicit def date2timestamp(date: java.util.Date): java.sql.Timestamp =
+    new java.sql.Timestamp(date.getTime)
+
+  // WdlValues have multiple ways they coerce back. This currently decoerces for a proper round trip.
+  // TODO: Refactor into the type?
+  private def wdlValueToString(wdlValue: WdlValue) = {
+    wdlValue match {
+      case wdlPrimitive: WdlPrimitive => wdlPrimitive.asString
+      case other => other.toString
+    }
+  }
+
+  // Will stamp the start_dt column with DateTime.Now.
+  // Will stamp the WorkflowState as Submitted.
+  override def createWorkflow(id: WorkflowId, wdlUri: String, symbols: Seq[SymbolStoreEntry]): Unit = {
+    val insertFuture: Future[Unit] = database.run((for {
+      workflowExecutionInsert <- dataAccess.insertWorkflowExecution(
+        new WorkflowExecution(id.toString, wdlUri, WorkflowSubmitted.toString, new Date())
+      )
+      symbolsInserts <- dataAccess.insertSymbols(
+        symbols map { symbol =>
+          new Symbol(
+            workflowExecutionInsert.workflowExecutionId.get,
+            symbol.key.scope, symbol.key.name, symbol.key.iteration,
+            if (symbol.isInput) "INPUT" else "OUTPUT",
+            symbol.wdlType.toWdlString, symbol.wdlValue.map(wdlValueToString).orNull)
+        }
+      )
+    } yield ()).transactionally)
+    // TODO: Return Future[Unit] instead of Unit?
+    Await.result(insertFuture, Duration.Inf)
+  }
+
+  override def updateCall(workflowId: WorkflowId, call: Call, notUsedByMethod: Option[CallStatus],
+                          callInfoOption: Option[CallInfo], symbolsOption: Option[Seq[SymbolStoreEntry]]): Unit = {
+    val workflowExecutionSelect = dataAccess.workflowExecutions.filter(
+      _.workflowExecutionUuid === workflowId.toString).result.head
+
+    val callInsertsOrUpdates = callInfoOption map { callInfo =>
```

Comment:
I prefer keeping combinators and cases either single calls or a small closure. Also if the interior blocks were narrowed down to simpler calls I suspect this will naturally start morphing into a for comprehension (as it's a `map` which contains a bunch of `flatMap`s)